src/pybind/mgr/nfs/ganesha_conf.py

```diff
@@ -558,3 +750,28 @@ def format_block(block: RawBlock, depth: int = 0) -> str:
     conf_str += _indentation(depth)
     conf_str += "}\n"
     return conf_str
+
+
+QOS_REQ_PARAMS = {
+    'combined_bw_disabled': {
+        'PerShare': ['max_export_write_bw', 'max_export_read_bw'],
+        'PerClient': ['max_client_write_bw', 'max_client_read_bw'],
+        'PerShare_PerClient': ['max_export_write_bw', 'max_export_read_bw', 'max_client_write_bw', 'max_client_read_bw']
+    },
+    'combined_bw_enabled': {
+        'PerShare': ['max_export_combined_bw'],
+        'PerClient': ['max_client_combined_bw'],
+        'PerShare_PerClient': ['max_export_combined_bw', 'max_client_combined_bw'],
+    }
+}
+
+
+def qos_bandwidth_checks(qos_type: QOSType, combined_bw_ctrl: bool, **kwargs: Any) -> None:
+    """Checks for enabling qos"""
+    if not combined_bw_ctrl:
+        req_params = QOS_REQ_PARAMS['combined_bw_disabled'][qos_type.value]
+    else:
+        req_params = QOS_REQ_PARAMS['combined_bw_enabled'][qos_type.value]
+    if any((key in req_params and kwargs[key] == '0')
+           or (key not in req_params and kwargs[key] != '0') for key in kwargs):
+        raise Exception(f"When combined_rw_bw is {'enabled' if combined_bw_ctrl else 'disabled'} and qos_type is {qos_type.value}, only the following parameters are required: {','.join(req_params)}.")
```

Comment:
Sort of a nitpick, but it might be worth separating these into two different error cases. One for missing parameters and one for invalid parameters. E.g. assume I have an operation with params A, B, C, and D and I both require and exclusively allow A and B in case 1 and C and D in case 2. To me, if I was told case 2 requires C and D, I would think I could pass in values for A, C, and D and it would be fine since I fulfilled the required values. In that case for an error I would hope for something like `attribute C is not allowed in case 2` and if I passed in just C to case 2, I'd hope for something like `attribute D is required in case 2`.